integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioCrossClusters.java

```diff
@@ -0,0 +1,467 @@
+
+  private static String K8S_NODEPORT_HOST1 = System.getenv("K8S_NODEPORT_HOST1");
+
+  private static boolean TWO_CLUSTERS = Boolean.parseBoolean(java.util.Optional.ofNullable(
+      System.getenv("TWO_CLUSTERS"))
+      .orElse("false"));
+
+  /**
+   * Install Operator.
+   * @param namespaces list of namespaces created by the IntegrationTestWatcher by the
+   *     JUnit engine parameter resolution mechanism
+   */
+  @BeforeAll
+  public static void initAll(@Namespaces(3) List<String> namespaces) {
+    logger = getLogger();
+    try {
+      FileInputStream in = new FileInputStream(PROPS_TEMP_DIR + "/" + WDT_MODEL_DOMAIN1_PROPS);
+      Properties props = new Properties();
+      props.load(in);
+      istioClusterOneIngressPort = props.getProperty("ISTIO_INGRESS_PORT");
+      in.close();
+    } catch (IOException ex) {
+      logger.info("Can't read property file");
+    }
+    //start to setup in cluster2
+    // create standard, reusable retry/backoff policy
+    withStandardRetryPolicy = with().pollDelay(2, SECONDS)
+        .and().with().pollInterval(10, SECONDS)
+        .atMost(5, MINUTES).await();
+    // get a new unique opNamespace
+    logger.info("Creating unique namespace for Operator");
+
+    assertNotNull(namespaces.get(0), "Namespace list is null");
+    op2Namespace = namespaces.get(0);
+
+
+    logger.info("Creating unique namespace for Domain");
+    assertNotNull(namespaces.get(2), "Namespace list is null");
+    domain2Namespace = namespaces.get(2);
+
+    dbUrl = ORACLEDBURLPREFIX + domain2Namespace + ORACLEDBSUFFIX;
+    createSecretForBaseImages(domain2Namespace);
+
+    //Start oracleDB
+    assertDoesNotThrow(() -> {
+      startOracleDB(DB_IMAGE_TO_USE_IN_SPEC, 0, domain2Namespace);
+      String.format("Failed to start Oracle DB");
+    });
+    dbNodePort = getDBNodePort(domain2Namespace, "oracledb");
+    logger.info("DB Node Port = {0}", dbNodePort);
+
+    // Label the domain/operator namespace with istio-injection=enabled
+    Map<String, String> labelMap = new HashMap();
+    labelMap.put("istio-injection", "enabled");
+
+    //assertDoesNotThrow(() -> addLabelsToNamespace(domain1Namespace,labelMap));
+    assertDoesNotThrow(() -> addLabelsToNamespace(domain2Namespace, labelMap));
+    assertDoesNotThrow(() -> addLabelsToNamespace(op2Namespace, labelMap));
+
+    // install and verify operator in cluster2
+    installAndVerifyOperator(op2Namespace, domain2Namespace);
+
+    // Now that we got the namespaces for both the domains, we need to update the model properties
+    // file with the namespaces. For cross domain transaction to work, we need to have the externalDNSName
+    // set in the config file. Cannot set this after the domain is up since a server restart is
+    // required for this to take effect. So, copying the property file to RESULT_ROOT and updating the
+    // property file
+    updatePropertyFile();
+  }
+
+  private static void createDomainAndApps() {
```

Comment:
This seems to start domain2. Can't see where you are starting domain1

Reply:
all related to domain1 is configured in ItIstioCrossClustersSetup.java

Reply:
Sorry, missed seeing that file. Is it too difficult to combine both in 1 java file?

Reply:
yes it is :(,  to create objects in different clusters you need to use different KUBECONFIG files for k8s client apis, I can't make a switch it in one java program, we load apiClient static block to create Kubernetes object, so even I change env var it loads kubeconfig only once.

Reply:
Fair enough. I figured KUBECONFIG would have to change, but wasn't sure  what else was the issue.